consensus/istanbul/backend/handler.go

```diff
@@ -253,9 +303,9 @@ func (sb *Backend) UpdateMetricsForParentOfBlock(child *types.Block) {
 }
 
 // Actions triggered by a new block being added to the chain.
-func (sb *Backend) NewChainHead(newBlock *types.Block) {
+func (sb *Backend) newChainHead(newBlock *types.Block) {
```

Comment:
From our conversation before, it looks that his method in not actually called for every chainHead, right? Or at least, not realtime.

We should review the contents of the method, probably we were assuming otherwise

Reply:
It's mainly metrics. On epoch transitions it does call refresh val peers for announce, but I think that gets called other places as well.

Reply:
If i'm not mistaken this method is not called for EVERY block. Depending on how blockchain blocks are inserted, it's batched. Which means we might "miss" en epoch block...

Can we check that? i'm worried that we untapped a problem here